test/unit/fact_importer_test.rb

```diff
@@ -27,5 +32,30 @@ class CustomImporter < FactImporter; end
     test 'importers without authorized_smart_proxy_features return empty set of features' do
       assert_equal [], FactImporter.importer_for(:custom_importer).authorized_smart_proxy_features
     end
+
+    context 'importing facts' do
+      setup do
+        disable_orchestration
+        User.current = users :admin
+        @host = FactoryGirl.create(:host)
+      end
+
+      test 'facts of other type do not collide even if they inherit from FactName' do
+        assert_nothing_raised do
+          custom_import '_timestamp' => '234'
```

Comment:
I wanted to remain consistent with https://github.com/theforeman/foreman/blob/develop/test/unit/puppet_fact_importer_test.rb#L91 maybe later with more tests it might be handy but if you want, I can remove it